File: sdks/sdkwork-mail-sdk/sdkwork-mail-sdk-swift/Sources/MailSdk/mailCapabilityCatalog.swift
public struct MailCapabilityCatalogEntry {
    public let capabilityKey: String
    public let category: String
    public let surface: String
}

public enum MailCapabilityCatalog {
    public static let entries: [MailCapabilityCatalogEntry] = [
        .init(capabilityKey: "transport.connect", category: "required-baseline", surface: "control-plane"),
        .init(capabilityKey: "transport.authenticate", category: "required-baseline", surface: "control-plane"),
        .init(capabilityKey: "health", category: "required-baseline", surface: "control-plane"),
        .init(capabilityKey: "smtp.send", category: "optional-advanced", surface: "runtime-bridge"),
        .init(capabilityKey: "imap.sync", category: "optional-advanced", surface: "runtime-bridge"),
        .init(capabilityKey: "imap.folder-sync", category: "optional-advanced", surface: "runtime-bridge"),
        .init(capabilityKey: "imap.message-sync", category: "optional-advanced", surface: "runtime-bridge"),
        .init(capabilityKey: "transport.retry", category: "optional-advanced", surface: "control-plane"),
        .init(capabilityKey: "transport.pool", category: "optional-advanced", surface: "control-plane"),
    ]

public static func getMailCapabilityCatalog() -> [MailCapabilityCatalogEntry] {
        entries
    }

    public static func getMailCapabilityDescriptor(_ capabilityKey: String) -> MailCapabilityCatalogEntry? {
        entries.first { $0.capabilityKey == capabilityKey }
    }

}

File: sdks/sdkwork-mail-sdk/sdkwork-mail-sdk-kotlin/src/main/kotlin/com/sdkwork/Mail/metadata/mailCapabilityCatalog.kt
package com.sdkwork.Mail.metadata

data class MailCapabilityCatalogEntry(
    val capabilityKey: String,
    val category: String,
    val surface: String,
)

object MailCapabilityCatalog {
    val entries: List<MailCapabilityCatalogEntry> = listOf(
        MailCapabilityCatalogEntry("transport.connect", "required-baseline", "control-plane"),
        MailCapabilityCatalogEntry("transport.authenticate", "required-baseline", "control-plane"),
        MailCapabilityCatalogEntry("health", "required-baseline", "control-plane"),
        MailCapabilityCatalogEntry("smtp.send", "optional-advanced", "runtime-bridge"),
        MailCapabilityCatalogEntry("imap.sync", "optional-advanced", "runtime-bridge"),
        MailCapabilityCatalogEntry("imap.folder-sync", "optional-advanced", "runtime-bridge"),
        MailCapabilityCatalogEntry("imap.message-sync", "optional-advanced", "runtime-bridge"),
        MailCapabilityCatalogEntry("transport.retry", "optional-advanced", "control-plane"),
        MailCapabilityCatalogEntry("transport.pool", "optional-advanced", "control-plane"),
    )

fun getMailCapabilityCatalog(): List<MailCapabilityCatalogEntry> = entries

    fun getMailCapabilityDescriptor(capabilityKey: String): MailCapabilityCatalogEntry? =
        entries.firstOrNull { it.capabilityKey == capabilityKey }

}
